api/src/paths/project/{projectId}/survey/{surveyId}/sample-site/delete.ts

```diff
@@ -0,0 +1,154 @@
+import { getLogger } from '../../../../../../utils/logger';
+
+const defaultLog = getLogger('paths/project/{projectId}/survey/{surveyId}/sample-site/delete');
+
+export const POST: Operation = [
+  authorizeRequestHandler((req) => {
+    return {
+      or: [
+        {
+          validProjectPermissions: [PROJECT_PERMISSION.COORDINATOR],
+          projectId: Number(req.params.projectId),
+          discriminator: 'ProjectPermission'
+        },
+        {
+          validSystemRoles: [SYSTEM_ROLE.DATA_ADMINISTRATOR],
+          discriminator: 'SystemRole'
+        }
+      ]
+    };
+  }),
+  deleteSurveySampleSiteRecords()
+];
+
+POST.apiDoc = {
+  description: 'Delete survey sample sites.',
+  tags: ['survey'],
+  security: [
+    {
+      Bearer: []
+    }
+  ],
+  parameters: [
+    {
+      in: 'path',
+      name: 'projectId',
+      schema: {
+        type: 'integer',
+        minimum: 1
+      },
+      required: true
+    },
+    {
+      in: 'path',
+      name: 'surveyId',
+      schema: {
+        type: 'integer',
+        minimum: 1
+      },
+      required: true
+    }
+  ],
+  requestBody: {
+    description: 'Survey sample site delete request object.',
+    content: {
+      'application/json': {
+        schema: {
+          type: 'object',
+          required: ['surveySampleSiteIds'],
+          properties: {
+            surveySampleSiteIds: {
+              items: {
+                type: 'integer',
+                minimum: 1
+              }
+            }
+          }
+        }
+      }
+    }
+  },
+  responses: {
+    204: {
+      description: 'Delete survey sample site OK'
+    },
+    400: {
+      $ref: '#/components/responses/400'
+    },
+    401: {
+      $ref: '#/components/responses/401'
+    },
+    403: {
+      $ref: '#/components/responses/403'
+    },
+    500: {
+      $ref: '#/components/responses/500'
+    },
+    default: {
+      $ref: '#/components/responses/default'
+    }
+  }
+};
+
+export function deleteSurveySampleSiteRecords(): RequestHandler {
+  return async (req, res) => {
+    const surveyId = Number(req.params.surveyId);
+    const surveySampleSiteIds = req.body.surveySampleSiteIds as number[];
+
+    if (!surveySampleSiteIds) {
+      throw new HTTP400('Missing required body `surveySampleSiteIds`');
+    }
+
+    const connection = getDBConnection(req['keycloak_token']);
+
+    try {
+      await connection.open();
+
+      const observationService = new ObservationService(connection);
+      const sampleLocationService = new SampleLocationService(connection);
+
+      const sampleLocations = await sampleLocationService.getSampleLocationsForSurveyId(surveyId);
+
+      const unableToDelete: number[] = [];
+
+      const deleteSampleSites = surveySampleSiteIds.map(async (surveySampleSiteId) => {
+        if (
+          (await observationService.getObservationsCountBySampleSiteId(surveyId, surveySampleSiteId)).observationCount >
```

Comment:
This could be really inefficent if I were deleting more than 10 sampling sites or so. Can't we write an SQL query that just finds all sample site IDs that are associated with one or more observations?

Reply:
Yes this is probably a better way of handling this delete call. Updated in next push.